packages/bundlers/experimental/src/ExperimentalBundler.js

```diff
@@ -788,6 +788,24 @@ function createIdealGraph(
         let sourceBundles = reachable.map(a => nullthrows(bundles.get(a.id)));
         let key = reachable.map(a => a.id).join(',');
         let bundleId = bundles.get(key);
+
+        sourceBundles.filter(sourceBundleId => {
```

Comment:
One thing to consider here is where we should actually just create all of the shared bundles, and then at the end, remove bundles from bundle groups that hit the limit. I did this in [my prototype](https://github.com/devongovett/bundler-algorithm/blob/master/src/main.rs#L204-L244). One benefit of that approach is that a shared bundle would still be created and used by bundle groups that are under the parallel request limit, and duplicated in those that are over it, rather than the shared bundle not being created at all. Also, we can do something smarter once we know all possible bundle groups, like remove the smallest bundles from the bundle groups over the limit, or the least frequently reused, for example.